### ISSUE TITLE
Referencing stack memory and a @frame pointer before suspending

### ISSUE THREAD

**Issue by jorangreef** [Sobeston/ziglearn#76]
Hey @Sobeston thanks for ziglearn! It's been a huge help learning Zig.

I had an idea to clarify the Async section.

Normally, passing a pointer to a struct allocated on the stack to another function somewhere else is verboten because it's UB.

However, with `suspend`, it seems like a common pattern is to do exactly this, with a struct allocated on the stack including some data as well as the @frame pointer, and then passing the @ptrToInt of the pointer to this struct somewhere else, e.g. as the user data when submitting an io_uring sqe.

It would be great to see an example of this, and an explanation that it's safe, and why.

**Comment by daurnimator** [comment]
> Normally, passing a pointer to a struct allocated on the stack to another function somewhere else is verboten because it's UB (the stack memory disappears and becomes undefined once the function returns).

Not exactly.

Something like this is entirely valid and normal:
```zig
fn foo() void {
   var buf: [100]u8 = undefined;
   bar(&buf);
}
```

The key thing is not `return`-ing a pointer to stack memory (or allowing the stack pointer out into the world for longer than after you `return`).

And `suspend` is not `return`.

**Comment by jorangreef** [comment]
> The key thing is not return-ing a pointer to stack memory (or allowing the stack pointer out into the world for longer than after you return).

Yes, thanks for clarifying, that was the UB I had in mind (the stack memory disappears and becomes undefined once the function returns).
